scripts/tls_test.c

```diff
@@ -402,7 +430,15 @@ int main(int argc, char *argv[])
 	cache = tls_cache_create(100, 30);
 	if (listen)
 	{
-		res = serve(host, server, times, cache, min_version, max_version);
+		if (cauth)
+		{
+			client = find_client_id();
```

Comment:
Not sure if this makes sense as the client certificate won't be loaded on the server, i.e. that function will only find the server identity. Maybe just assign ID_ANY (`identification_create_from_encoding(ID_ANY, chunk_empty);`).  You also wouldn't need the `cauth` flag. Instead assign that identity to `client`, which you initialize to `NULL`, above in the arg parse loop.

Reply:
Thanks, this looks better.